client/landing/stepper/declarative-flow/internals/steps-repository/site-creation-step/index.tsx

```diff
@@ -55,7 +55,11 @@ const SiteCreationStep: Step = function SiteCreationStep( { navigation, flow, da
 
 	const username = useSelector( ( state ) => getCurrentUserName( state ) );
 
-	const { setPendingAction } = useDispatch( ONBOARD_STORE );
+	const { setPendingAction, setPluginsToVerify } = useDispatch( ONBOARD_STORE );
+
+	if ( isWooExpressFlow( flow ) ) {
+		setPluginsToVerify( [ 'woocommerce' ] );
```

Comment:
Should we not run this logic from the flow itself? It might make sense to do it just before we redirect to the `waitForPluginInstall` step.

Having this logic embedded in the step doesn't feel right to me at all!

Reply:
I was going from your earlier comment:

> We could persist the set of plugins at the start of a flow _(or during any step before this one)_, and simply have this step verify that a list of plugin slugs have been installed on the site.

and so I put it in the first step of the flow. But I can definitely move it to the flow itself if you think that makes more sense.

Reply:
From my side it's more about the separation of concerns than about sequencing, if that makes sense. Having the _step_ responsible for logic that is flow-specific feels like it could easily be fragile. I know we have lots of other logic doing that already, but I think we should aim to minimise that as much as possible.

Reply:
That all makes total sense. I'll get that updated.